steam_helper: separate parameters from command when using ShellExecuteW

Jan200101 opened the pull request:
ShellExecuteW ignores any extra information passed as part of lpFile
resulting in arguments being ignored.

GloriousEggroll commented:
been testing this for a while, unfortunately it breaks a few games:

Guilty Gear Xrd
Escape from Monkey Island
Memento Mori

Jan200101 committed:
steam_helper: separate parameters from command when using ShellExecuteW

ShellExecuteW ignores any extra information passed as part of lpFile
resulting in arguments being ignored.

Jan200101 commented:
rebased onto 8.0 and fixed the issue mentioned by GloriousEggroll tested using Guilty Gear Xrd REV 2 and Ultrakill to ensure nothing else broke.

To go into more issue what caused this issue:
GGXrd uses a bat to launch which is started using an absolute path by Steam
steam_helper converts that absolute unix path into an absolute dos path and quotes it
the logic within `get_end_of_excutable_name` respects this and stops iterating the cmdline once it finds the end of quote
This leaves our pointer dangling at the end quote instead of after the program

I updated the `get_end_of_excutable_name` function to go to the full end, quotes included, and checking for a quote within `should_use_shell_execute`.

GloriousEggroll commented:
Confirming everything working as expected now, thanks @Jan200101

ishitatsuyuki commented:
@ivyl This seems ready for review, and according to the user is necessary for a Titanfall 2 mod (Northstar) to work. Would you mind taking a look?
